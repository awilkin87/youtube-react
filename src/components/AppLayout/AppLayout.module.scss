@use '../../styles/shared.module';

.appLayout {
  & > *:not(:first-child) {
    margin-top: shared.$header-nav-height;
  }

  ::-webkit-scrollbar {
    width: 8px;
  }
}