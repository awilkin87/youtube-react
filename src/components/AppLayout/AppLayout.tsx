import HeaderNav from '../../containers/HeaderNav/HeaderNav';
import styles from './AppLayout.module.scss';

/** Main application layout, placing the header nav bar at the top.  */
const AppLayout: React.FunctionComponent<Readonly<{}>> = function (props) {
  return (
    <div className={styles.appLayout}>
      <HeaderNav />
      {props.children}
    </div>
  );
}

export default AppLayout;