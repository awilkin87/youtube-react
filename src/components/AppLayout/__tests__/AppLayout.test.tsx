import { shallow } from 'enzyme'
import AppLayout from '../AppLayout'

describe('AppLayout', () => {
  test('Render AppLayout with no childern', () => {
    const wrapper = shallow(<AppLayout />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Render AppLayout with no child', () => {
    const wrapper = shallow(
      <AppLayout>
        <div>Child 1</div>
      </AppLayout>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Render AppLayout with two childern', () => {
    const wrapper = shallow(
      <AppLayout>
        <div>Child 1</div>
        <div>
          <p>Child 2A</p>
          <span>Child 2B</span>
        </div>
      </AppLayout>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
