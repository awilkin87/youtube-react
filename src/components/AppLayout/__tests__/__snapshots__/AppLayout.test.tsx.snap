// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout Render AppLayout with no child 1`] = `
<div
  className="appLayout"
>
  <HeaderNav />
  <div>
    Child 1
  </div>
</div>
`;

exports[`AppLayout Render AppLayout with no childern 1`] = `
<div
  className="appLayout"
>
  <HeaderNav />
</div>
`;

exports[`AppLayout Render AppLayout with two childern 1`] = `
<div
  className="appLayout"
>
  <HeaderNav />
  <div>
    Child 1
  </div>
  <div>
    <p>
      Child 2A
    </p>
    <span>
      Child 2B
    </span>
  </div>
</div>
`;
